core/slider.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2016 Massachusetts Institute of Technology
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The slider class used for field sliders
+ * @author Chase Mortensen
+ */
+'use strict';
+
+goog.provide('Blockly.Slider');
+    
+Blockly.Slider = function() {
+
+    // Set up default values
+    this.step_ = 1;
+
+    this.slider = this;
+    this.createDom();
+    this.setupEvents();
+    
+}
+
+Blockly.Slider.prototype.setupEvents = function() {
+    Blockly.bindEvent_(this.thumb_, "mousedown", this, function(ev) {
+        if (this && this.slider) {
+            this.slider.mouseDown_ = true;
+            if (this.slider.onClickEvent) {
+                this.slider.onClickEvent();
+            }
+        }
+    })
+
+    Blockly.bindEvent_(this.element_, "mousedown", this, function(ev) {
+        if (this && this.slider) {
+            this.slider.mouseDown_ = true;
+            if (this.slider.onClickEvent) {
+                this.slider.onClickEvent();
+            }
+            if (this.moveToPoint) {
+                this.updatePosition(ev.clientX);
+            }
+        }
+    })
+
+    Blockly.bindEvent_(window, "mouseup", this, function (ev) {
```

Comment:
If you bind an event on the window, it needs to get cleaned up somewhere. Typically we bind these events in the mousedown and mouseup handlers, so the flow looks something like this:

1. mousedown fires
2. In the event handler, register the mouseup and mousemove events on the window
3. mouseup fires
4. In the event handler, remove the mouseup and mousemove handlers from the window